### PR TITLE
[PAN-2072] handle root configuration path

### PR DESCRIPTION
**Pull request by danut13**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

- Fix the issue where the service was not able to load the config in case of a working directory which points to the root
- Relax the provider timeout typing: allow None
- Add the blockchain nodes domains to the error message instead of the URI itself
- Add a thread pool for checking the health of the nodes

### CODE REVIEW

**Review by markuslevonyak — DISMISSED** [review]

**Comment by jacekv** [comment]
Address the code smell please: https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&id=pantos-io_common&open=AYs-etUPS24f8mvx8ULq

**Comment by markuslevonyak** [comment]
> Address the code smell please: https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&id=pantos-io_common&open=AYs-etUPS24f8mvx8ULq

By the way, it would be great to see the SonarCloud report directly in the PR again.

**Comment by jacekv** [comment]
Let's have a look what happened here regarding SonarCloud @markuslevonyak

**Comment by danut13** [comment]
@jacekv that's not a code smell this PR introduced. Actually, that analysis does not look like it's performed on this PR, but on the main branch. I'm not sure what's going on with Sonar, doesn't look like it's working properly

**Comment by jacekv** [comment]
@danut13 but is the code smell also part of your code? If so, you can fix it, even you haven't introduced it. I am looking into getting SonarCloud to post messages again, so that we do not overlook those problems :)

**Review by markuslevonyak — APPROVED** [review]

File: pantos/common/configuration.py [hunk 1]
```diff
@@ -19,13 +19,15 @@
 
 # Ordered by priority
 _CONFIGURATION_PATHS = [
-    pathlib.Path.cwd(),
     pathlib.Path.home(),
     pathlib.Path.home() / '.config',
     pathlib.Path('/etc/pantos'),
     pathlib.Path('/etc')
 ]
```
Comment (chandraprakash) [inline pantos/common/configuration.py:26]: Alternative
```
_CONFIGURATION_PATHS = [ pathlib.Path.cwd() if pathlib.Path.cwd() != pathlib.Path('/') else None ] + [
    pathlib.Path.home(),
    pathlib.Path.home() / '.config',
    pathlib.Path('/etc/pantos'),
    pathlib.Path('/etc')
]
```

**Review by chandraprakash — APPROVED** [review]

File: pantos/common/configuration.py (hunk 1 above)
Comment (danut13) [reply pantos/common/configuration.py:26]: Hmm, any option is ugly IMO, I will just merge like this 👍

**Pull request merged**